File: viikko1/c++/helloworld/Opiskelija.cpp
#include "Opiskelija.h"
#include <iostream>
using namespace std;

Opiskelija::Opiskelija() : mNimi("tuntematon"), mOpiskelijanumero("0000"), mOpintopisteet(0)
{
    cout << "Opiskelijan oletusrakentaja" <<endl;
}

Opiskelija::Opiskelija(string aNimi, string aOpiskelijanumero)
: mNimi(aNimi), mOpiskelijanumero(aOpiskelijanumero), mOpintopisteet(0)
{
    cout <<"Opiskelijan 2 parametrinen rakentaja" <<endl; 
}

Opiskelija::~Opiskelija()
{
    cout <<"Opiskelija " << mNimi << " vapautettu muistista" << endl;
}

void Opiskelija::setNimi(string aNimi)
{
    mNimi = aNimi;  
}

void Opiskelija::setOpintipisteet(unsigned short int aOpintopisteet)
{
    mOpintopisteet = aOpintopisteet;
}

void Opiskelija::setOpiskelijanumero(string aOpiskelijanumero)
{
    mOpiskelijanumero = aOpiskelijanumero;
}

string Opiskelija::getNimi()
{
    return mNimi;
}

string Opiskelija::getOpiskelijanumero()
{
    return mOpiskelijanumero;
}

unsigned short int Opiskelija::getOpintopisteet()
{
    return mOpintopisteet;
}

void Opiskelija::tulostaTiedot()
{
 cout<<"Nimi: " << mNimi << endl;
 cout<<"Opiskelijanumero: " << mOpiskelijanumero <<endl;
 cout<<"Opintopisteet: " << mOpintopisteet <<endl;
}

File: viikko2/java/Opiskelija.java
public class Opiskelija {
    // DATA
    private String mNimi;
    private String mOpiskelijanumero;
    private int mOpintiopisteet;

    public Opiskelija(){
        mNimi = "tuntematon";
        mOpiskelijanumero = "xxxxxx";
        mOpintiopisteet = 10;
    }
    public Opiskelija( String aNimi, String aOpiskelijanumero){
        mNimi = aNimi;
        mOpiskelijanumero = aOpiskelijanumero;
    }
    public void setNimi( String aNimi){
        mNimi = aNimi;
    }
    public void setOpintopisteet( int aOpintopisteet){
        mOpintiopisteet = aOpintopisteet;
    }
    public void setOpiskelijanumero( String aOpiskelijanumero){
        mOpiskelijanumero = aOpiskelijanumero;
    }

    public String getNimi(){
        return mNimi;
    }

    public String getOpiskelijanumero(){
        return mOpiskelijanumero;
    }

    public int getOpintopisteet(){
        return mOpintiopisteet;
    }
    public void lisaaKurssi(int aOpintopisteet){
        mOpintiopisteet = aOpintopisteet;
    }


    public void tulostaTiedot() {
        System.out.println("Nimi: " + mNimi);
        System.out.println("Opiskelijanumero: " + mOpiskelijanumero);
        System.out.println("Opintopisteet: " + mOpintiopisteet);
    }

}